### PR TITLE
fix: meta not respecting social variables, customized title

### PR DESCRIPTION
**Pull request by ccorda**
refs #174
refs #172
refs #177

### CODE REVIEW

File: website/src/components/Meta.js [hunk 1]
```diff
@@ -1,43 +1,52 @@
...
   const globals = useContext(GlobalsContext);
 
+  // make sure image is absolute
+  function imagePath(imageUrl) {
+    // if relative, make absolute so that twitter doesn't complain
+    if (
+      imageUrl.indexOf('http://') === -1 &&
+      imageUrl.indexOf('https://') === -1
+    ) {
+      imageUrl = META.url + imageUrl;
```
Comment (chrisherold) [inline website/src/components/Meta.js:17]: is it possible there's a situation where `imageUrl` is relative and without a leading slash and we need to add a slash between `META.url` and `imageUrl`? Also should we expect users/devs to always set the `META.url` without a trailing slash or do we need to check/sanitize that as well?

**Review by chrisherold — APPROVED** [review]
one note on image url generation, otherwise lgtm

File: website/src/components/Meta.js (hunk 1 above)
Comment (ccorda) [reply website/src/components/Meta.js:17]: good feedback, ty. agree we should check/document.

**Comment by ccorda** [comment]
Made those change @chrisherold. I realized that the image proxy i implemented was an 80% solution, but missing the actual solve (the issue is that twitter won't index images when you have robots.txt that blocks access, which WP engine does on .wpengine domains). I made a new config variable, and the redirect rule so that next proxies the remote image.

Confirmed this is output:

`<meta property="og:image" content="https://bubs.patronage.org/wp-content/uploads/2021/10/spacejoy-EVjqpcn79AM-unsplash.jpg"/>`

That path then proxies WP, but does so without robot blocking metas that make twitter mad.

Also setup our test case to have custom Twitter meta fields and confirmed they work:
https://bubs-next-f7yb6eq92-patronage.vercel.app/test-cases/test-custom-social/

**Review by kylehotchkiss — APPROVED** [review]

**Review by chrisherold — APPROVED** [review]
lgtm

**Pull request merged**